Integrations/QRadar/QRadar.py

```diff
@@ -1038,6 +1048,116 @@ def get_domains_by_id_command():
+        else:
+            if element_type:
+                return_error("The reference set {0} is already exist. You are not supposed to enter element type,"
+                             " Try again".format(reference_name))
+        query = args.get('query')
+        value_indicators_list, indicators_data_list = get_indicators_list(query)
+        if len(value_indicators_list) == 0:
+            return {
+                'Type': entryTypes['note'],
+                'Contents': {},
+                'ContentsFormat': formats['json'],
+                'ReadableContentsFormat': formats['markdown'],
+                'HumanReadable': "No indicators found, Reference set {0} didn't change".format(reference_name)
+            }
+        else:
+            raw_response = upload_indicators_list_request(reference_name, value_indicators_list)
+            ref_set_data = unicode_to_str_recur(get_reference_by_name(reference_name))
+            indicator_headers = ['value', 'indicator_type']
+            ref_set_headers = ['name', 'element_type', 'timeout_type', 'creation_time', 'number_of_elements']
+            return {
+                'Type': entryTypes['note'],
+                'Contents': raw_response,
+                'ContentsFormat': formats['json'],
+                'ReadableContentsFormat': formats['markdown'],
+                'HumanReadable':
+                    tableToMarkdown("reference set {0} was updated".format(reference_name), ref_set_data,
+                                    headers=ref_set_headers) + tableToMarkdown("Indicators list",
+                                                                               indicators_data_list,
+                                                                               headers=indicator_headers)}
+    except Exception as e:
+        if '1005' in str(e):
+            return "You tried to add indicators that dont match to reference set type"
+        raise e
+
+
+def check_ref_set_exist(ref_set_name):
+    """
+        The function checks if reference set is exist
+
+    Args:
+        ref_set_name (str): Reference set name
+
+    Returns:
+        Dict. If found - Reference set object, else - Error
+    """
+
+    try:
+        return get_reference_by_name(ref_set_name)
+    except Exception as e:
+        if '1002' in str(e):
+            return None
+        raise e
+
+
+def get_indicators_list(indicator_query):
+    """
+        Get Demisto indicators list using demisto.searchIndicators
+
+        Args:
+              indicator_query (str): The query demisto.searchIndicators use to find indicators
+        Returns:
+             list of indicators value and a list with all indicators data
+    """
+    last_found_len = PAGE_SIZE
+    value_indicators_list = []
+    indicators_data_list = []
+    page = 0
+    while last_found_len == PAGE_SIZE:
+        fetched_iocs = demisto.searchIndicators(query=indicator_query, page=page, size=PAGE_SIZE).get('iocs')
```

Comment:
we need pagination here, otherwise the user will get stuck with the limit he entered